apis/sgv2-quarkus-common/src/main/java/io/stargate/sgv2/api/common/logging/LoggingFilter.java

```diff
@@ -0,0 +1,147 @@
+
+import static io.stargate.sgv2.api.common.config.constants.LoggingConstants.*;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import io.stargate.sgv2.api.common.StargateRequestInfo;
+import io.stargate.sgv2.api.common.config.LoggingConfig;
+import jakarta.enterprise.context.ApplicationScoped;
+import jakarta.inject.Inject;
+import jakarta.ws.rs.container.ContainerRequestContext;
+import jakarta.ws.rs.container.ContainerResponseContext;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Collections;
+import java.util.Set;
+import org.jboss.resteasy.reactive.server.ServerResponseFilter;
+import org.slf4j.Logger;
+
+/** The filter for logging requests when required for debugging purposes. */
+@ApplicationScoped
+public class LoggingFilter {
+  private static final Logger logger = org.slf4j.LoggerFactory.getLogger(LoggingFilter.class);
+
+  /** The {@link ObjectMapper} to get request body from request. */
+  private final ObjectMapper objectMapper;
+  /** The request info bean. */
+  private final StargateRequestInfo requestInfo;
+  /** The configuration for logging. */
+  private final LoggingConfig loggingConfig;
+
+  /** Default constructor. */
+  @Inject
+  public LoggingFilter(
+      ObjectMapper objectMapper, StargateRequestInfo requestInfo, LoggingConfig loggingConfig) {
+    this.objectMapper = objectMapper;
+    this.requestInfo = requestInfo;
+    this.loggingConfig = loggingConfig;
+  }
+
+  /**
+   * Filter that this bean produces.
+   *
+   * @param requestContext {@link ContainerRequestContext}
+   * @param responseContext {@link ContainerResponseContext}
+   * @see https://quarkus.io/guides/resteasy-reactive#request-or-response-filters
+   */
+  @ServerResponseFilter
+  public void logRequestIfRequired(
+      ContainerRequestContext requestContext, ContainerResponseContext responseContext) {
+    if (isLoggingAllowed(requestContext, responseContext)) {
+      logger.debug(
+          buildRequestInfo(
+              requestContext, responseContext, loggingConfig.requestBodyLoggingEnabled()));
+    }
+  }
+
+  private String buildRequestInfo(
+      ContainerRequestContext requestContext,
+      ContainerResponseContext responseContext,
+      boolean logRequestBody) {
+    String requestBody = logRequestBody ? getRequestBody(requestContext) : "";
+    return String.format(
+        "REQUEST INFO :: %s %s %s %s %s",
+        requestInfo.getTenantId().orElse("tenant-unknown"),
+        responseContext.getStatus(),
+        requestContext.getMethod(),
+        requestContext.getUriInfo().getPath(),
+        requestBody);
+  }
+
+  private String getRequestBody(ContainerRequestContext requestContext) {
+    try {
+      InputStream inputStream = requestContext.getEntityStream();
+      if (inputStream == null || !inputStream.markSupported()) {
+        return "";
+      }
+      inputStream.reset();
+      return objectMapper.readTree(inputStream).toString();
```

Comment:
What is the point of decoding JSON, then re-encoding? Wouldn't it be easier to just read contents as String (assuming UTF-8 encoding)?

Reply:
Good catch. I originally thought of pretty printing the request body, which is available in the `readTree` method of `ObjectMapper`. Then I changed it to simple `toString` as it was logged in multiline which will make debugging harder if splunk parses them as separate lines. Let me change this to simple read from `InputStream` as string.

Reply:
Fixed in [f13a3eb](https://github.com/stargate/stargate/pull/2754/commits/f13a3eb82770bfaa7dc902f694f4285946909fe9)